hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/index/columnstats/ColumnStatsIndexHelper.java

```diff
@@ -447,8 +448,8 @@ private static String composeZIndexColName(String col, String statName) {
           new Float(colMetadata.getMaxValue().toString()));
     } else if (colType instanceof BinaryType) {
       return Pair.of(
-          ((Binary) colMetadata.getMinValue()).getBytes(),
-          ((Binary) colMetadata.getMaxValue()).getBytes());
+          ((ByteBuffer) colMetadata.getMinValue()).array(),
```

Comment:
Now that `convertToNativeJavaType()` returns a java String type for ParquetUTF8BinaryStringType, Line 426 needs to change right? No need for Binary casting there.